dev/tools/test-e2e

```diff
@@ -14,31 +14,128 @@
 # limitations under the License.
 
 import os
+import argparse
 import subprocess
 import sys
 
 from shared import utils
 
 
-def main():
+def run_go_e2e_tests(repo_root, artifact_dir, env):
+    """Runs the Go e2e tests"""
+    print("========= Running Go E2E tests... =========")
+    go_test_cmd = utils.go_tool_args(
+        "gotestsum",
```

Comment:
Not a blocker, but I don't think gotestsum does much, and I think we can do much better.  Not a big deal until it takes > 10 minutes for a test run :-)